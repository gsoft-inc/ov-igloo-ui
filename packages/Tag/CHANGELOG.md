# @igloo-ui/tag

## 1.5.2

### Patch Changes

- a91a24d: Updated the ghost secondary color of the icon button since it differs from the ghost secondary button color.
- Updated dependencies [a91a24d]
  - @igloo-ui/icon-button@1.2.4

## 1.5.1

### Patch Changes

- 97edb0e: The button is now square when there's only an icon showing on mobile.
- Updated dependencies [97edb0e]
  - @igloo-ui/icon-button@1.2.3

## 1.5.0

### Minor Changes

- bea2c50: Added the "select" appearence for the Tag that is used inside TagPicker. Fixed height and spacing of TagPicker.

## 1.4.1

### Patch Changes

- 6ea531f: Updated dependency versions for react-aria packages, luxon and hopper
- Updated dependencies [6ea531f]
  - @igloo-ui/ellipsis@0.3.1
  - @igloo-ui/icon-button@1.2.2

## 1.4.0

### Minor Changes

- e2cc8b8: Added the "select" appearence for the Tag that is used inside TagPicker. Fixed height and spacing of TagPicker.

## 1.3.2

### Patch Changes

- 8bf2e79: Version bump to acknowledge button patch version
- Updated dependencies [8bf2e79]
  - @igloo-ui/icon-button@1.2.1

## 1.3.1

### Patch Changes

- 0c419dc: Fix an issue where classNames were not passed to children when cloned

## 1.3.0

### Minor Changes

- 4846c59: Wrapped Igloo CSS in @layer to isolate it.

### Patch Changes

- Updated dependencies [4846c59]
  - @igloo-ui/ellipsis@0.3.0
  - @igloo-ui/icon-button@1.2.0

## 1.2.5

### Patch Changes

- cfb5619: Updated hopper token versions and button versions.
- Updated dependencies [cfb5619]
  - @igloo-ui/ellipsis@0.2.4
  - @igloo-ui/icon-button@1.1.7

## 1.2.4

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.
- Updated dependencies [7d36815]
  - @igloo-ui/ellipsis@0.2.3
  - @igloo-ui/icon-button@1.1.6

## 1.2.3

### Patch Changes

- 5082846: added space between icon and border for round tags small, xsmall and micro

## 1.2.2

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency
- Updated dependencies [a2df55d]
  - @igloo-ui/ellipsis@0.2.2
  - @igloo-ui/icon-button@1.1.5

## 1.2.1

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.
- Updated dependencies [9739155]
  - @igloo-ui/ellipsis@0.2.1
  - @igloo-ui/icon-button@1.1.4

## 1.2.0

### Minor Changes

- 09c0000: update tag to change border-radius to match new brand

## 1.1.1

### Patch Changes

- Updated dependencies [36fbdcc]
  - @igloo-ui/ellipsis@0.2.0
  - @igloo-ui/icon-button@1.1.3

## 1.1.0

### Minor Changes

- 5402bcc: Created a "brand" toggle in the Toolbar and applied the new Workleap brand to some components. Also deprecated the Tag's appearance values and added new ones.

### Patch Changes

- Updated dependencies [5402bcc]
  - @igloo-ui/icon-button@1.1.0

## 1.0.8

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.
- Updated dependencies [bf4ddef]
  - @igloo-ui/ellipsis@0.1.11
  - @igloo-ui/icon-button@1.0.4

## 1.0.7

### Patch Changes

- c3efaf5: Fixed tests and snapshots since react-aria generated ids changed and viewBox was added to icons.
- Updated dependencies [c3efaf5]
  - @igloo-ui/icon-button@1.0.3

## 1.0.6

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.
- Updated dependencies [a41e1d4]
  - @igloo-ui/ellipsis@0.1.10
  - @igloo-ui/icon-button@1.0.1

## 1.0.5

### Patch Changes

- 55d12e3: fix error with empty package @shared/components

## 1.0.4

### Patch Changes

- 4230499: Fix: updated components to use the IconButton with appearance ghost, variant secondary so that they are grey like they used to be.

## 1.0.3

### Patch Changes

- d5c91f5: link shared components with aliases
- Updated dependencies [e47ae88]
  - @igloo-ui/icon-button@1.0.0

## 1.0.2

### Patch Changes

- 9b9ec61: Initial release of the tag picker and a small color change for the dismiss button in the tag component.

## 1.0.1

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console
- Updated dependencies [6ee9c45]
  - @igloo-ui/ellipsis@0.1.8
  - @igloo-ui/icon-button@0.2.13

## 1.0.0

### Major Changes

- 2a6d8e5: Extracted the Color and Avatar components from the Tag component

  ## BREAKING CHANGE

  Restructuring HTML in the tag component as well as renaming onClose to OnRemove since it made more sense. onRemove also gains an id as a parameter.

## 0.2.0

### Minor Changes

- 8d0b6ed: Added new features such as an ellipsis when too long, a color icon option, an image icon option, a warning icon option, and an error look

## 0.1.2

### Patch Changes

- 0640409: Update @igloo-ui/icons to 1.6.0
- 0640409: udpate classnames to 2.3.2
- Updated dependencies [0640409]
- Updated dependencies [0640409]
  - @igloo-ui/icon-button@0.2.12

## 0.1.1

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme
- Updated dependencies [d971143]
  - @igloo-ui/icon-button@0.2.11

## 0.1.0

### Minor Changes

- 5eeb191: Initial release of tag component
