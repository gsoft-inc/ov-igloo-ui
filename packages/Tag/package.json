{
  "name": "@igloo-ui/tag",
  "version": "1.0.6",
  "main": "dist/Tag.js",
  "module": "dist/Tag.js",
  "types": "dist/Tag.d.ts",
  "styles": "dist/tag.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Vicky Comeau vicky.comeau@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/icon-button": "^1.0.1",
    "@igloo-ui/icons": "^1.11.0",
    "@igloo-ui/tokens": "^2.0.0",
    "@igloo-ui/ellipsis": "^0.1.10",
    "classnames": "^2.3.2"
  },
  "devDependencies": {
    "@shared/components": "0.1.1"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
