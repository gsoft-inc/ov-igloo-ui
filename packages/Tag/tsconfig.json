{
  "extends": "../../tsconfig.build.json",
  "include": ["src/**/*", "src/intl/*.json", "../../typings/**/*", "rollup.config.js"],
  "exclude": ["dist"],
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist",
    "composite": true,
    "baseUrl": ".",
    "typeRoots": ["../../typings", "../../node_modules/@types"]
  },
  "references": [
    {
      "path": "../../shared/components"
    }
  ]
}
