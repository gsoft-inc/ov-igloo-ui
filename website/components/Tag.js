import cx from 'classnames';

export default function Tag({ children, className, appearance }) {
  return (
    <span
      className={cx('io-tag', className, {
        [`io-tag--${appearance}`]: appearance,
      })}
    >
      {children}
    </span>
  );
}
