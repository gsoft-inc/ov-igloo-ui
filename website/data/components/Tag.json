[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Tag/src/Tag.tsx",
      "description": "",
      "displayName": "Tag",
      "methods": [],
      "props": {
        "appearance": {
          "defaultValue": { "value": "default" },
          "description": "The different appearances of the Tag",
          "name": "appearance",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            }
          ],
          "required": false,
          "type": { "name": "Appearance" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "children": {
          "defaultValue": null,
          "description": "The content of the Tag",
          "name": "children",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "className": {
          "defaultValue": null,
          "description": "Add a specific class to the Tag",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "dismissible": {
          "defaultValue": { "value": false },
          "description": "Render the close button",
          "name": "dismissible",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "icon": {
          "defaultValue": null,
          "description": "The icon used at the beginning of the Tag",
          "name": "icon",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "onClose": {
          "defaultValue": null,
          "description": "Event when the tag is closed",
          "name": "onClose",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            }
          ],
          "required": false,
          "type": { "name": "() => void" }
        },
        "rounded": {
          "defaultValue": { "value": false },
          "description": "Render rounded corners",
          "name": "rounded",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "size": {
          "defaultValue": { "value": "medium" },
          "description": "The different sizes of the Tag",
          "name": "size",
          "parent": {
            "fileName": "components/packages/Tag/src/Tag.tsx",
            "name": "TagProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tag/src/Tag.tsx",
              "name": "TagProps"
            }
          ],
          "required": false,
          "type": { "name": "Size" }
        }
      }
    }
  ]
]
