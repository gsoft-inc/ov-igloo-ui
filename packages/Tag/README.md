# Tag

The Tag component informs users of the status of an object or of an action that's been taken.

<Example />

<ReferenceLinks />

## Installation

To install `@igloo-ui/tag` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/tag
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/tag
```

## Usage

Then to use the component in your code just import it!

```jsx
import Tag from '@igloo-ui/tag';

<Tag icon={<LabelSolid size="small" />} appearance="success" dismissible>
  This is a tag
</Tag>;
```
