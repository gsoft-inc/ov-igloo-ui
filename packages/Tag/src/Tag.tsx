import * as React from "react";

import cx from "classnames";

// eslint-disable-next-line import/no-extraneous-dependencies
import { VisualIdentifier } from "@shared/components";
import IconButton from "@igloo-ui/icon-button";
import Ellipsis from "@igloo-ui/ellipsis";
import Close from "@igloo-ui/icons/dist/Close";

import "./tag.scss";

export type Size = "medium" | "small" | "xsmall" | "micro";

export type AppearanceNew =
  | "default"
  | "primary"
  | "progress"
  | "positive"
  | "caution"
  | "negative"
  | "neutral";

/** @Deprecated This will be removed Nov. 1, 2023 */
export type AppearanceDep =
  | "info"
  | "success"
  | "warning"
  | "error"
  | "secondary"
  | "grey";

export type Appearance = AppearanceNew | AppearanceDep;

export interface TagProps extends React.ComponentProps<"div"> {
    /** The different appearances of the Tag */
    appearance?: Appearance;
    /** The content of the Tag */
    children: React.ReactNode;
    /** Add a specific class to the Tag */
    className?: string;
    /** Add a colored square instead of an image or an icon */
    color?: string;
    /** Add a data-test tag for automated tests */
    dataTest?: string;
    /** Render the close button */
    dismissible?: boolean;
    /** The icon used at the beginning of the Tag */
    icon?: React.ReactElement;
    /** The id of the tag to use when removing */
    id?: string;
    /** Add an error style to the tag */
    hasError?: boolean;
    /** Callback to execute on remove tag */
    onRemove?: (id: string) => void;
    /** Render rounded corners */
    rounded?: boolean;
    /** The different sizes of the Tag */
    size?: Size;
    /** Specifies the image url to show */
    src?: string;
}

const Tag: React.FunctionComponent<TagProps> = ({
    children,
    className,
    color,
    dataTest,
    dismissible = false,
    appearance = "default",
    icon,
    id,
    hasError,
    onRemove,
    rounded = false,
    size = "medium",
    src
}) => {
    const [show, setShow] = React.useState(true);

    const printDeprecationWarning = (oldValue: string, newValue: string): void => {
        console.warn(`Warning: Tag's '${oldValue}' appearance is deprecated and will be removed ` +
        `Nov. 1, 2023. Use '${newValue}' instead. `);
    };

    const mapDeprecatedAppearance = (appearanceValue: Appearance): AppearanceNew => {
        switch (appearanceValue) {
            case "info":
                printDeprecationWarning("info", "progress");

                return "progress";
            case "success":
                printDeprecationWarning("success", "positive");

                return "positive";
            case "warning":
                printDeprecationWarning("warning", "caution");

                return "caution";
            case "error":
                printDeprecationWarning("error", "negative");

                return "negative";
            case "grey":
                printDeprecationWarning("grey", "default");

                return "default";
            case "secondary":
                printDeprecationWarning("secondary", "neutral");

                return "neutral";
            default:
                return appearanceValue as AppearanceNew;
        }
    };

    const renderDismissButton = (): JSX.Element => {
        const action = (): void => {
            if (onRemove && id) {
                onRemove(id);
            }

            setShow(false);
        };

        return (
            <IconButton
                className="ids-tag__dismiss-btn"
                type="button"
                onClick={action}
                appearance={{ type: "ghost", variant: "secondary" }}
                size="xsmall"
                aria-label="close"
                icon={<Close size="small" className="ids-tag__close" />}
            />
        );
    };

    const classes = cx(
        "ids-tag",
        `ids-tag--${mapDeprecatedAppearance(appearance)}`,
        `ids-tag--${size}`,
        {
            "ids-tag--rounded": rounded,
            "ids-tag--has-error": hasError
        },
        className
    );

    if (show) {
        return (
            <div className={classes} data-test={dataTest}>
                <VisualIdentifier
                    className="ids-tag__visual"
                    color={color}
                    src={src}
                    icon={icon}
                />

                <div className="ids-tag__content">
                    <Ellipsis>{children}</Ellipsis>
                </div>

                {dismissible && renderDismissButton()}
            </div>
        );
    }

    return null;
};

export default Tag;
