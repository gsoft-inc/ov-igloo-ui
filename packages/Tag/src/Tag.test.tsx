/**
 * @jest-environment jsdom
 */
import React from 'react';
import {
  fireEvent,
  render,
  RenderResult,
  screen,
} from '@testing-library/react';

import Tag, { Appearance, Size, TagProps } from './Tag';
import LabelSolid from '@igloo-ui/icons/dist/LabelSolid';

const setup = (props: TagProps) => {
  return render(<Tag dataTest="ids-tag" {...props} />);
};

describe('Tag', () => {
  const component = (children: React.ReactNode) => setup({ children });

  const expectToBeOfStyle = (component: RenderResult, style: Appearance) => {
    const tag = component.container.firstChild;
    const expectedClass = 'ids-tag--' + style;
    expect(tag).toHaveClass(expectedClass);
  };

  const expectToBeOfSize = (component: RenderResult, size: Size) => {
    const tag = component.container.firstChild;
    const expectedClass = 'ids-tag--' + size;
    expect(tag).toHaveClass(expectedClass);
  };

  test('It should render without errors', () => {
    component('Render without errors');

    expect(screen.getByTestId('ids-tag')).toBeInTheDocument();
  });

  test('It should render a snapshot', () => {
    const { asFragment } = component('Render snapshot');
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should render by default a default style medium tag and no dismiss button', () => {
    const tag = setup({ children: 'Default Tag' });
    const findDismissButton = tag.container.querySelector(
      '.ids-tag__dismiss-btn'
    );

    expect(findDismissButton).not.toBeInTheDocument();

    expectToBeOfStyle(tag, 'default');
    expectToBeOfSize(tag, 'medium');
  });

  /* onClose function */

  test('It should render a tag with an onClose action', () => {
    let show = true;
    const props = {
      children: 'Test onClose Tag',
      dismissible: true,
      onClose: () => {
        show = false;
      },
    };
    const { container } = setup(props);

    const dismissBtn = container.querySelector('.ids-tag__dismiss-btn');
    if (dismissBtn) {
      fireEvent.click(dismissBtn);
    }
    expect(show).toBeFalsy();
  });

  /* Dismissible */

  test('It should render a tag with a dismiss button', () => {
    const { container } = setup({
      children: 'Dismissible Tag',
      dismissible: true,
    });
    const findDismissButton = container.querySelector('.ids-tag__dismiss-btn');

    expect(findDismissButton).toBeInTheDocument();
  });

  test('It should render a tag without a dismiss button', () => {
    const { container } = setup({
      children: 'Not Dismissible Tag',
      dismissible: false,
    });
    const findDismissButton = container.querySelector('.ids-tag__dismiss-btn');

    expect(findDismissButton).not.toBeInTheDocument();
  });

  /* Icon */

  test('It should render a tag without an icon', () => {
    const { container } = setup({ children: 'Default Icon Tag' });
    const icon = container.querySelector('.ids-tag__icon');

    expect(icon).not.toBeInTheDocument();
  });

  test('It should render a tag with an icon', () => {
    const { container } = setup({
      children: 'Dismissible Tag',
      icon: <LabelSolid size="small" />,
    });
    const icon = container.querySelector('.ids-tag__icon');

    expect(icon).toBeInTheDocument();
  });

  test('It should render a tag with an icon', () => {
    const { container } = setup({
      children: 'Tag with icon',
      icon: <LabelSolid size="small" />,
    });
    const icon = container.querySelector('.ids-tag__icon');

    expect(icon).toBeInTheDocument();
  });

  test('It should render a tag with a color icon', () => {
    const { container } = setup({
      children: 'Tag with color icon',
      color: '#9A3842',
    });
    const icon = container.querySelector('.ids-tag__color-icon');

    expect(icon).toBeInTheDocument();
  });

  test('It should render a tag with a color icon from ID', () => {
    const { container } = setup({
      children: 'Tag with color icon',
      color: '',
      id: '3434-3434',
    });
    const icon = container.querySelector('.ids-tag__color-icon');

    expect(icon).toBeInTheDocument();
  });

  test('It should render a tag with an image icon', () => {
    const { container } = setup({
      children: 'Tag with color icon',
      src: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQEB3tCC4oJsa4ZZsiEDIhWi06EyN3iHYMoOg&usqp=CAU',
    });
    const icon = container.querySelector('.ids-tag__image-icon');

    expect(icon).toBeInTheDocument();
  });

  test('It should render a tag with a warning icon', () => {
    const { container } = setup({
      children: 'Tag with color icon',
      isWarning: true,
    });
    const icon = container.querySelector('.ids-tag__warning-icon');

    expect(icon).toBeInTheDocument();
  });

  /* Appearance */

  test('It should render a default style tag', () => {
    const tag = setup({ children: 'Default Tag', appearance: 'default' });

    expectToBeOfStyle(tag, 'default');
  });

  test('It should render a primary style tag', () => {
    const tag = setup({ children: 'Primary Tag', appearance: 'primary' });

    expectToBeOfStyle(tag, 'primary');
  });

  test('It should render a secondary style tag', () => {
    const tag = setup({ children: 'Secondary Tag', appearance: 'secondary' });

    expectToBeOfStyle(tag, 'secondary');
  });

  test('It should render a info style tag', () => {
    const tag = setup({ children: 'Info Tag', appearance: 'info' });

    expectToBeOfStyle(tag, 'info');
  });

  test('It should render a success style tag', () => {
    const tag = setup({ children: 'Success Tag', appearance: 'success' });

    expectToBeOfStyle(tag, 'success');
  });

  test('It should render a warning style tag', () => {
    const tag = setup({ children: 'Warning Tag', appearance: 'warning' });

    expectToBeOfStyle(tag, 'warning');
  });

  test('It should render a error style tag', () => {
    const tag = setup({ children: 'Error Tag', appearance: 'error' });

    expectToBeOfStyle(tag, 'error');
  });

  /* Size */

  test('It should render a medium tag', () => {
    const tag = setup({ children: 'Medium Tag', size: 'medium' });

    expectToBeOfSize(tag, 'medium');
  });

  test('It should render a small tag', () => {
    const tag = setup({ children: 'Small Tag', size: 'small' });

    expectToBeOfSize(tag, 'small');
  });

  test('It should render a xsmall tag', () => {
    const tag = setup({ children: 'XSmall Tag', size: 'xsmall' });

    expectToBeOfSize(tag, 'xsmall');
  });

  test('It should render a micro tag', () => {
    const tag = setup({ children: 'Micro Tag', size: 'micro' });

    expectToBeOfSize(tag, 'micro');
  });

  test('It should render with an error class', () => {
    setup({
      children: 'Tag with an error',
      dataTest: 'ids-tag-error',
      appearance: 'secondary',
      hasError: true,
    });

    const tag = screen.getByTestId('ids-tag-error');
    expect(tag).toHaveClass('ids-tag--has-error');
  });
});
