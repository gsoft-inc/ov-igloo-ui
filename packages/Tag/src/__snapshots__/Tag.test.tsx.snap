// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag It should render a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ids-tag ids-tag--default ids-tag--medium"
    data-test="ids-tag"
  >
    <div
      class="ids-tag__content"
    >
      <div
        class="ids-ellipsis"
      >
        <div
          class="ids-ellipsis__content"
        >
          Render snapshot
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
